### PR TITLE
Implement Crypto Host Capability

### LINKED ISSUE #179: Feature Request: Implement the `v1/is_certificate_trusted` host capability
### Acceptance criteria

- Implement the `v1/is_certificate_trusted` capability in the SDK, following the WaPC protocol described in the [cryptographic capabilities spec](https://docs.kubewarden.io/reference/spec/host-capabilities/crypto).
- Add unit tests to validate the correct construction of the request payload, parsing of the host response, and edge cases
- Add integration tests against a mock Wasm host
- Add end-to-end tests via the test policy (using `kwctl` and `bats`) to verify that the capability behaves correctly when given trusted and untrusted certificates.

### PR DESCRIPTION
**Pull request by esosaoh**
## Description

Fix #179

This PR implements the `v1/is_certificate_trusted` host capability as specified in [here](https://docs.kubewarden.io/reference/spec/host-capabilities/crypto)

## Test

To test this pull request, you can run the following commands:

```shell
make e2e-tests
```

### CODE REVIEW

File: js/index.ts [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
+let pluginPath: string | undefined;
+try {
+  // this will only work in Node.js, not in WASM
+  const path = require('path');
+  pluginPath = path.resolve(__dirname, '../plugin/javy-plugin-kubewarden.wasm');
+} catch {
+  // in WASM environment, pluginPath remains undefined
+  pluginPath = undefined;
+}
+
```
[the remark this answers is not in the document]
Comment (esosaoh) [reply js/index.ts]: I've rebased this branch @jvanz

File: js/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "kubewarden-policy-sdk",
-  "version": "0.1.0-alpha.0",
```
[the remark this answers is not in the document]
Comment (viccuad) [reply js/package.json:3]: indeed, we should go with `alpha.1` to follow the already existing `0.1.0-alpha.0`.

**Comment by esosaoh** [comment]
@jvanz changed to a non-ca cert

File: js/kubewarden/host_capabilities/crypto/types.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * Certificate encoding types
+ */
+export type CertificateEncoding = 'Der' | 'Pem';
+
+/**
+ * Certificate represents a x509 certificate
+ */
+export interface Certificate {
+  encoding: CertificateEncoding;
+  data: number[];
```
Comment (viccuad) [inline js/kubewarden/host_capabilities/crypto/types.ts:11]: `data` must always be utf8, we should document it on the type here.

In addition, maybe we should provide the following functions:

```js
export namespace CertificateUtils {
  export function fromString(certString: string, encoding: CertificateEncoding): Certificate {
    return {
      encoding,
      data: Array.from(new TextEncoder().encode(certString)), // always UTF-8
    };
  }

  export function toString(cert: Certificate): string {
    return new TextDecoder('utf-8').decode(Uint8Array.from(cert.data));
  }
}
```

File: js/kubewarden/host_capabilities/crypto/types.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+/**
+ * CertificateVerificationRequest holds information about a certificate and
+ * a chain to validate it with
+ */
+export interface CertificateVerificationRequest {
+  cert: Certificate;
+  cert_chain: Certificate[];
```
Comment (viccuad) [inline js/kubewarden/host_capabilities/crypto/types.ts:20]: Both `cert_chain` and `not_after` are optional, hence they may be missing from the JSON that a policy sends to the host.

It seems that the `cert_chain` may be a None or an empty list, but for the `not_after` we need to not provide it to skip it.

File: js/kubewarden/host_capabilities/crypto/types.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+export interface Certificate {
+  encoding: CertificateEncoding;
+  data: number[];
+}
+
+/**
+ * CertificateVerificationRequest holds information about a certificate and
+ * a chain to validate it with
+ */
+export interface CertificateVerificationRequest {
```
Comment (viccuad) [inline js/kubewarden/host_capabilities/crypto/types.ts:18]: We don't need to export this, it's a helper internal type.
This also allows us to build the Request as needed, given the optional `cert_chain` and `not_after` that can be missing.

**Review by flavio — DISMISSED** [review]

File: js/kubewarden/host_capabilities/crypto/types.ts (hunk 5 above)
Comment (esosaoh) [reply js/kubewarden/host_capabilities/crypto/types.ts]: done but i had to exclude this line from the linter since the interface isn't used

**Review by viccuad — APPROVED** [review]
Looks good to me, thanks!

**Comment by flavio** [comment]
@esosaoh can you please fix the commit signatures? Otherwise, we won't be able to merge this PR
